src/backend/cpu/memory.hpp

```diff
@@ -53,13 +56,14 @@ void setMemStepSize(size_t step_bytes);
 size_t getMemStepSize(void);
 bool checkMemoryLimit();
 
-class MemoryManager : public common::MemoryManager<cpu::MemoryManager> {
+class MemoryManager : public BackendMemoryClient {
```

Comment:
A couple of comments about the class names. We can address these towards the end, just leaving a comment so that I don't forget about mentioning it.

- `common::MemoryManagerBase` and `common::MemoryManager` sounds good.
- `backend::MemoryManager` is confusing - I understand it has been like that before your changes. I would say , lets change it for less confusion.

Perhaps `common::BackendMemoryClient` -> `common::NativeMemoryInterface` ? This came naturally to me. 

But the next one, I am still confused as to what is better than `cpu/cuda/opencl::MemoryManager` ?

Reply:
`common::NativeMemoryInterface` is a lot better - I'll make that change.

We can use `cpu/cuda/opencl::NativeMemoryInterface`? I'll change it to that and see how it looks.